pkg/checks/checks.go

```diff
@@ -0,0 +1,249 @@
+package checks
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	appclient "k8s.io/client-go/kubernetes/typed/apps/v1"
+	coreclient "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// WaitForHTTPStatusOk poll until URL returns 200
+func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urltocheck string) error {
+	cli := &http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	req, err := http.NewRequest("GET", urltocheck, nil)
+	if err != nil {
+		return err
+	}
+
+	for {
+		resp, err := cli.Do(req)
+		if err, ok := err.(*url.Error); ok && (err.Timeout() || err.Err == io.EOF || err.Err == io.ErrUnexpectedEOF) {
+			time.Sleep(time.Second)
+			continue
+		}
+
+		if err == io.EOF {
+			time.Sleep(time.Second)
+			continue
+		}
+		if err != nil {
+			log.Printf("the err was %#v\n", err)
+			return err
+		}
+		if resp.StatusCode == http.StatusOK {
+			return nil
+		}
+
+		select {
+		case <-ctx.Done():
+			log.Printf("ctx.Done called.  Returning err")
+			return ctx.Err()
+		case <-time.After(time.Second):
+		}
+	}
+}
+
+func checkNamespace(namespace string) bool {
+	whitelistedNamespaces := []string{
+		"openshift-",
+		"kube-",
+		"default",
+	}
+	for _, ns := range whitelistedNamespaces {
+		if strings.HasPrefix(namespace, ns) {
+			return true
+		}
+	}
+	return false
+}
+
+// WaitForInfraServices verify daemonsets, statefulsets
```

Comment:
We probably need a hardcoded list of the expected workloads and cross-checking readiness vs mere listing. I am fine with a follow-up issue.

Reply:
Yeah, maybe that's enough

Reply:
Although I am worried about a missing workload for whatever reason, that's why I think a whitelist of the workloads is eventually required.